### DISCUSSION TITLE
Is there any special reason for packeto buildpack to use liberica for the jdk?

### DISCUSSION THREAD

**Discussion by heesuk-ahn** [paketo-buildpacks/feedback#15] [Q&A] [answered]
Hi all, I'm packeto newbie. :) 

Recently I am making an image through gradle plugin of spring boot.

But I found out that paketo builder uses liberica buildpack for jre internally.

Is there any particular reason why you're using this? I'm curious 🤔 

Thanks for reading the article! It would be a big help if you reply!

**Comment by fg-j** [comment]
Hey @paketo-buildpacks/java-buildpacks -- do you have thoughts about this?

I know that in the Paketo Java documentation we have some explanation about how to use an alternative JVM (Check out https://paketo.io/docs/howto/java). The example in the "Alternative JVMs" section uses the pack CLI, not the gradle plugin with spring boot, though.

**Comment by dmikusa** [comment, marked as answer]
I'm not 100% sure why/how we landed on using Liberica as the default. 

I can say the following though:

1. We needed a default, so we had to pick one.
2. There is a corporate entity behind Liberica, i.e. Bellsoft, which provides some additional stability (i.e. there are folks paid to work on it).
3. Bellsoft and the developers they employ have a strong history in the OpenJDK community
4. Bellsoft Liberica has consistently provided timely updates after upstream OpenJDK releases (they are often one of the first to release out of all our providers).
5. We treat the JVM vendor as an implementation detail. It's a compatible JVM, so most users won't notice a difference or need to care.
6. Some JVMs vendors offer additional features/functionality or some power users may have a preference, and for those users that wish to use a specific JVM vendor they can [switch the JVM vendor very easily](https://paketo.io/docs/howto/java/#using-alternative-jvms). 
7. You can also package up your own builder very easily if, for example, your company requires a different JVM vendor. You can take our builder, swap out Bellsoft and plug in your vendor of choice. We're committed to offering JVM vendor options and supporting as many different JVM vendors as we can reasonablly do.

**Comment by heesuk-ahn** [comment]
Thanks for the replies, it's been a big help! 👍 

In particular, the best thing is that we can flexibly replace the java buildpack through the builder pipeline! 💯 

I will also look for an option to change the java buildpack in the gradle plugin currently used by spring boot.
Thanks for your help :)

**Reply by dmikusa** [reply]
You can use [this documented approach](https://paketo.io/docs/howto/java/#using-alternative-jvms) with Spring Boot 2.5+ and the `buildpacks` [customization option](https://docs.spring.io/spring-boot/docs/2.5.3/maven-plugin/reference/htmlsingle/#build-image.customization).

You can also use a custom builder with the `builder` [customization option](https://docs.spring.io/spring-boot/docs/2.5.3/maven-plugin/reference/htmlsingle/#build-image.customization). This option has been around longer, so it would work if you're still on Spring Boot 2.4.

**Reply by heesuk-ahn** [reply]
I added the option `build.gradle` and it was very easy to see replacing it with the JVM I wanted. :) 

<img width="919" alt="스크린샷 2021-08-18 오전 10 43 11" src="https://user-images.githubusercontent.com/42938941/129823879-6afd3a9c-7752-4b0c-9c2b-67d600504fe6.png">

<img width="1188" alt="스크린샷 2021-08-18 오전 10 49 05" src="https://user-images.githubusercontent.com/42938941/129823950-e61d3f42-0ea3-4e60-8ae7-8591a7112920.png">
